### DISCUSSION TITLE
how to solve this problem?

### DISCUSSION THREAD

**Discussion by aafnnp** [pmndrs/zustand#1227] [Q&A]
```js
import create from 'zustand'
import { devtools, persist } from 'zustand/middleware'

interface CartState {
	total: number
	totalqty: number
	cartContent: []
	addTocart: (params) => void
	updatecart: (params) => void
	clearCart: (params) => void
	removeFromCart: (params) => void
}
const useCart = create<CartState, [['zustand/devtools', never], ['zustand/persist', CartState]]>(
	devtools(
		persist(
			set => ({
				total: 0,
				totalqty: 0,
				cartContent: [],
				addTocart: params => {
					set(state => ({
						totalqty: state.totalqty + 1,
						total: state.total + parseFloat(params.price),
						cartContent: [...state.cartContent, params],
					}))
				},
				updatecart: ({ params, mycart }) => {
					set(state => ({
						totalqty: state.totalqty + 1,
						total: state.total + parseFloat(params.price),
						cartContent: mycart,
					}))
				},
				clearCart: () => set({ totalqty: 0, total: 0, cartContent: [] }),
				removeFromCart: params =>
					set(state => ({
						total: state.total - params.price * params.quantity,
						totalqty: state.totalqty - params.quantity,
						cartContent: state.cartContent.filter(item => item.id !== params.id),
					})),
			}),
			{ name: 'cart', getStorage: () => sessionStorage }
		)
	)
)

export default useCart
```

Argument of type '(state: CartState) => { totalqty: number; total: number; cartContent: [any]; }' is not assignable to parameter of type 'CartState | Partial<CartState> | ((state: CartState) => CartState | Partial<CartState>)'.
  Type '(state: CartState) => { totalqty: number; total: number; cartContent: [any]; }' is not assignable to type '(state: CartState) => CartState | Partial<CartState>'.
    Type '{ totalqty: number; total: number; cartContent: [any]; }' is not assignable to type 'CartState | Partial<CartState>'.
      Type '{ totalqty: number; total: number; cartContent: [any]; }' is not assignable to type 'Partial<CartState>'.
        Types of property 'cartContent' are incompatible.
          Type '[any]' is not assignable to type '[]'.
            Source has 1 element(s) but target allows only 0.

**Comment by dai-shi** [comment]
Would you be able to create a minimal reproduction with https://www.typescriptlang.org/play/ ?
